mdbenchmark/tests/test_utils.py

```diff
@@ -0,0 +1,96 @@
+import os
+from glob import glob
+
+import click
+import numpy as np
+from mdbenchmark.utils import (calc_slope_intercept, cleanup_before_restart,
+                               get_possible_hosts, guess_host, lin_func,
+                               normalize_host)
+
+import pytest
+
+
+def test_get_possible_hosts():
+    """Assert that we can retrieve the correct number of hosts.
+    These are determined by the job files currently shipped
+    with this package.
+    """
+    hosts = get_possible_hosts()
+    assert isinstance(hosts, list)
+    assert len(hosts) == 2
+
+
+def test_guess_host():
+    """Assert that `guess_host()` does not recognize a random machine as a host.
+
+    Note: this test may fail on a machine that, by accident, is called like a
+    host file that we provide.
+    """
+    assert guess_host() is None
+
+
+def test_monkeypatch_guess_host(monkeypatch):
+    """Assert that `guess_host()` will correctly assign the hostname to a host
+    file.
+    """
+    host = 'HPC_cluster'
+    monkeypatch.setattr('mdbenchmark.utils.socket.gethostname', lambda: host)
+    monkeypatch.setattr('mdbenchmark.utils.get_possible_hosts', lambda: [host])
+    assert guess_host() == host
+
+
+def test_normalize_host(monkeypatch):
+    """Test `normalize_host()`."""
+    with pytest.raises(click.BadParameter):
+        normalize_host(None)
+
+    host = 'HPC_cluster'
+    monkeypatch.setattr('mdbenchmark.utils.guess_host', lambda: host)
+    assert normalize_host(host) == host
+
+
+def test_lin_func():
+    """Test `lin_func()`."""
+    m, x, b = [5, 3, 2]
+
+    np.testing.assert_array_almost_equal(lin_func(m, x, b), (m * x) + b)
+
+
+def test_calc_slope_intercept():
+    """Test `calc_slope_intercept()`"""
+    x1, y1 = [1, 1]
+    x2, y2 = [2, 2]
+    slope = (y2 - y1) / (x2 - x1)
+    intercept = y1 - (x1 * slope)
+
+    slope_intercept = calc_slope_intercept(x1, y1, x2, y2)
+
+    np.testing.assert_array_almost_equal(slope_intercept,
```

Comment:
use top level import of assert function. Also `assert_almost_equal` is more strict then the `assert_array_*` functions. Don't ask me why numpy devs did this.